src/ex12_html_css_js/task_02.js

```diff
@@ -0,0 +1,48 @@
+window.onload = function () {
+
+var topMenu = document.getElementById("top-menu");
+var menuItems = document.getElementsByClassName("menu-item");
```

Comment:
why don't you call `Array.from` once right here?

Reply:
corrected

Reply:
I meant this approach `var menuItemsArray = Array.from(document.getElementsByClassName('menu-item'));`

Reply:
var menuItems = document.getElementsByClassName("menu-item") => menuItems variable already contains an array of items. I used the Array method.from as an alternative to the for loop for hanging events

Reply:
Actually, it's not an array, but [HTMLCollection](https://developer.mozilla.org/docs/Web/API/HTMLCollection), that's hasn't support of Array.prototype methods, that's why you're using `Array.from` there, but if you would really like to use `Array.from` as a replacement for `looping` that isn't good actually, because it has another purpose -- to convert an array-like object to an array, but not for simple iteration on that

Reply:
and then why convert several times if once is enough

Reply:
Yes, because of this I was not able to use forEach, because it is not suitable for the collection. i.e the entry will be like this:
 var menuItemsArray = Array.from(menuItems, function (elem) {
    elem.addEventListener("mouseover", setBg);
    elem.addEventListener("mouseout", setBg);
});
?

Reply:
do it in 2 steps:
1. extract the collection of elements and coerce it to an array
2. add event listeners with a common #forEach approach